Implemented new reviewer chooser and delete dead code

kaedroho opened the pull request.

emilytoppm committed:
Remove code removing 'submit for moderation' option

emilytoppm committed:
Begin refactor of Wagtail Review to work with the upcoming Wagtail workflow feature: converting review request references to task state references, and adding a new ReviewTask and ReviewTaskState

emilytoppm committed:
Add task action functionality from review frontend.

kaedroho committed:
Implemented new reviewer chooser

kaedroho committed:
Delete dead code

kaedroho committed:
Reorder actions

So that "review" is the default action

emilytoppm commented on wagtail_review/admin_api/serializers.py:
```diff
@@ -1,16 +1,36 @@
 
+class UserSerializer(serializers.ModelSerializer):
+    id = serializers.ReadOnlyField(source='pk')
+    full_name = serializers.ReadOnlyField(source='get_full_name')
```
Just noting that this is optional for a custom user model, as per https://docs.djangoproject.com/en/3.0/topics/auth/customizing/. Is this relied upon?

kaedroho replied:
Fixed in latest commit

emilytoppm commented on wagtail_review/static_src/admin/src/api/reviewer.ts:
```diff
@@ -0,0 +1,103 @@
+
+    async _newReviwer(body: string): Promise<NewReviewerResponse> {
+        let response = await fetch(`/admin/wagtail_review/api/reviewers/`, {
+            method: 'POST',
```
The `AdminAPIViewMixin` relies on session authentication, which according to https://www.django-rest-framework.org/api-guide/authentication/#sessionauthentication will require a valid CSRF token for any POST requests as here. Is this attached somehow? Or does `FooAuth` circumvent this, in which case there's maybe a case for this being somewhat unsafe?

kaedroho replied:
The fact it's called `FooAuth` suggests to me this was only a temporary hack that I've forgotten about. Thanks for spotting!

kaedroho replied:
Fixed in latest commit

emilytoppm reviewed (changes requested):
Generally looks great, but there's a few issues remaining (partly my fault, as we handed over mid-development!). 
Should be fixed: 
- `allow_responses` in the template tag is inconsistent with the `Respond` view: potentially allowing a reviewer to pick an action they don't actually have permission to perform in the API
- Tests need to be updated and run
To look at:
- UserSerializer
- CSRF issues

kaedroho committed:
Set CSRF token on POST requests

kaedroho committed:
Use single name field for user

That takes get_full_name but falls back to get_username if that's not
implemented

kaedroho committed:
Add TravisCI config

kaedroho committed:
Bump version to 0.3 alpha

kaedroho committed:
Drop support for Wagtail 2.5

kaedroho committed:
Delete admin tests

All these views have now been replaced with APIs

kaedroho committed:
Fix tests

kaedroho committed:
Reword decided_at/decided_by fields

kaedroho committed:
Use ReviewerPagePermissions for checking if responses are allowed

gasman reviewed (approved):
Latest round of fixes all look good to me!

kaedroho committed:
Styling comments in editor

kaedroho committed:
Remove use of removed Page.dummy_request method

kaedroho committed:
Rename TaskState fields and implement get_comment

kaedroho committed:
Commit new static bundles

gasman commented on wagtail_review/models.py:
```diff
@@ -345,23 +347,51 @@ class ReviewResponse(models.Model):
+                    <blockquote>{comment}</blockquote>
+                """)
+            else:
+                return mark_safe(f"<blockquote>{comment}</blockquote>")
```
Is there a reason why this isn't using Django's `format_html`? It seems that would eliminate a lot of the explicit escaping and mark_safe (and avoid the python-3.5 incompatibility of format strings).

kaedroho replied:
Just didn't think of doing it that way, will change now. Thanks!

kaedroho committed:
Change PostgreSQL port number back to 5432 on Travis CI

https://travis-ci.community/t/postgres-default-port-changed-from-5432-to-5433/7347/5
